packages/@glimmer/compiler-delegates/src/module-unification/compiler-delegate.ts

```diff
@@ -62,9 +66,15 @@ export default class MUCompilerDelegate implements AppCompilerDelegate<TemplateM
    */
   templateLocatorFor({ module, name }: ModuleLocator): TemplateLocator<TemplateMeta> {
     let relativePath = module.replace(/^\.\//, '');
-    let meta = {
-      specifier: this.project.specifierForPath(relativePath)
-    };
+
+    let meta;
+    if (name === 'mainTemplate') {
```

Comment:
…for example, why are we hardcoding the check for `mainTemplate` here? Shouldn't we just check to see if the locator passed is in `builtins` first and skip normalization in those cases, like we do with helpers?

IOW, the semantics should be: locators in `builtins` are ambient and not normalized, anything else is assumed to come from the project and need normalization.